### PR TITLE
Update Uri.php

### PR DESCRIPTION
**Pull request by vicn1222**
Pull Request for Issue #42123 .

### Summary of Changes

This change is to fix Joomla router unable to parse custom SEF url. The SEF url has a rewrite rule defined in .htaccess. where the SEF url need to be converted to dynamic url with query string.

The affected function is Uri::getInstance() in libraries/src/Uri directory.

### Testing Instructions

1. create an article "foo-article" and a menu "foo-menu". The "foo-menu" will accept query string, such as

  q=xyz&dir=abc&name=ijk

2. The browser url will look like http://localhost/foo-menu/xyz/abc/ijk.html

3. In .htaccess, add re-write rule below

RewriteRule ^foo-menu/(.*)/(.*)/(.*).html$ index.php/foo-menu?q=$1&dir=$2&name=$3 [NC,L]

4. According to the .htaccess, the browser url needs be converted to

http://localhost/foo-menu?q=xyz&dir=abc&name=ijk

5. The new url can then be passed into the Joomla router for menu entry lookup, resulting in correct parameters being sent to the menu for further processing.

### Actual result BEFORE applying this Pull Request

The .htaccess rewrite rule is not applied prior to calling Joomla's router. A custom router is required.

### Expected result AFTER applying this Pull Request

The rewrite rule in the ,htaccess is the custom router, dynamic url is generated before calling Joomla router. No extra custom router is needed.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ x] No documentation changes for manual.joomla.org needed

### Code changes
Code changes have been uploaded as "Uri.txt"  in the issue tracker at https://issues.joomla.org/tracker/joomla-cms/42123

### CODE REVIEW

**Comment by vicn1222** [comment]
Tests have been done in J3 & J4 multiple times. I am just unsure if I made the right pull request. This is the first time I am using github. So please correct me if there is any error in the pull request. I have been using svn all the time.

The change should also be applied to J3 if there is any possibility.

Thank you for the wonderful project!

-Frank

File: libraries/src/Uri/Uri.php [hunk 1]
```diff
@@ -83,7 +83,12 @@ public static function getInstance($uri = 'SERVER')
                  * are present, we will assume we are running on apache.
                  */
 
-                if (!empty($_SERVER['PHP_SELF']) && !empty($_SERVER['REQUEST_URI'])) {
+                if (!empty($_SERVER['PHP_SELF']) && !empty($_SERVER['PATH_INFO']) && !empty($_SERVER['QUERY_STRING'])) {
+
+                    // Convert browser url to uqery string url according to ReWrite rule
+                    $theURI = 'http' . $https . $_SERVER['HTTP_HOST'] . $_SERVER['PATH_INFO'] . '?' . $_SERVER['QUERY_STRING'];
+
```
Comment (richard67) [inline libraries/src/Uri/Uri.php]:
```suggestion
```
Remove empty line to fix PHP code style check failing in Drone CI.

**Comment by Quy** [comment]
@vicn1222 Welcome! Thank you for your contribution.

FYI: To see drone issues, please go to the bottom of the page. Click `Details`. Click  `phpcs` to see the issues to fix.

drone

**Comment by vicn1222** [comment]
Failure was due to "else if". It should be "elseif"...

Look like this checkin is going to 4.4-dev. Will it automatically propagates to later version, such as 5.x?

I am totally new to github...

File: libraries/src/Uri/Uri.php (hunk 1 above)
Comment (vicn1222) [reply libraries/src/Uri/Uri.php]: Done

**Review by vicn1222 — COMMENTED** [review]
Done

**Pull request closed without merging**